src/editors/containers/LibraryContentEditor/index.test.jsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+import { LibraryContentEditor } from './index';
+
+jest.unmock('@edx/paragon');
+jest.unmock('@edx/paragon/icons');
+
+jest.mock('./LibrarySettings', () => jest.fn(
+  props => <div {...props}>LibrarySettings</div>,
+));
+jest.mock('./LibrarySelector', () => jest.fn(
+  props => <div {...props}>LibrarySelector</div>,
+));
+jest.mock('./BlocksSelector', () => jest.fn(
+  props => <div {...props}>BlocksSelector</div>,
+));
+jest.mock('../EditorContainer', () => jest.fn(
+  props => <div name="EditorContainer" {...props} />,
+));
+
+jest.mock('./hooks', () => ({
+  useLibraryHook: jest.fn().mockReturnValue({
+    getContent: jest.fn(),
+  }),
+}));
+
+function renderComponent(props) {
+  return render(
+    <IntlProvider locale="en">
+      <LibraryContentEditor {...props} />
+    </IntlProvider>,
+  );
+}
+
+describe('LibraryContentEditor', () => {
+  const props = {
+    onClose: jest.fn(),
+    returnFunction: jest.fn(),
+    blockFailed: false,
+    blockFinished: true,
+    blockValue: 'SoMe VaLue',
+    blocksInSelectedLibrary: ['block1', 'block2', 'block3'],
+    candidates: ['block3'],
+    libraryPayload: { payload: 'pAYloAD' },
+    intl: {
+      formatMessage: (content) => content.defaultMessage,
+    },
+  };
+
+  it('Renders a spinner when loading', () => {
+    const { getByTestId } = renderComponent({ ...props, blockFinished: false });
+    expect(getByTestId('librarycontenteditor-loadingspinner')).toBeTruthy();
+  });
+  it('Renders as expected once loaded', () => {
```

Comment:
Also on this component level, we need to consider what should happen if the loading has finished but the block fails. What should be rendered? Should child components be rendered if there is no block to render? Should we throw an error that NewRelic will pick up? - If we do still render child components, will child components throw unwanted errors?

I would include a test for it with the desired behavior, and probably the desired behavior warrants some thinking about what should happen.

- That's my first-glance take, maybe you already dealt with everything in the hook. It wouldn't hurt having an additional test here anyway, so we can see if a future change inside the component breaks something.

Reply:
I think we want whatever is the most simple here in terms of behavior @rayzhou-bit , but +1 to having a unit test here.